lib/src/values/random_logic_value.dart

```diff
@@ -50,28 +59,31 @@ extension RandLogicValue on math.Random {
     } else {
       if (width <= LogicValue._INT_BITS) {
         LogicValue ranNum;
-        if (width < 32) {
+        if (width <= 32) {
           ranNum = LogicValue.ofInt(nextInt(1 << width), width);
         } else {
           ranNum = LogicValue.ofInt(_nextBigInt(numBits: width).toInt(), width);
         }
 
-        return LogicValue.ofInt(
-            ranNum.toInt() % (max == null ? (1 << width) - 1 : max.toInt()),
-            width);
+        if (max == null) {
+          return ranNum;
+        } else {
+          if (max is BigInt) {
+            throw InvalidRandomLogicValueException(
+                'input width limit only allows Max to be type Int.');
+          } else {
+            max = max as int;
+          }
+          return LogicValue.ofInt(ranNum.toInt() % max, width);
+        }
       } else {
-        var ranNum = _nextBigInt(numBits: width);
+        final ranNum = _nextBigInt(numBits: width);
 
-        if (max == null || ranNum.bitLength < max.width) {
+        if (max == null ||
+            (max is BigInt && ranNum.bitLength < max.bitLength)) {
```

Comment:
Maybe we need to be careful with `bitLength`, since I think negative numbers act differently than positive numbers (e.g. -1 bitlength is 1 i think, even though it's 0xffffffff)?  I wonder if we should try to make some mocks or something to test specific outputs from the random generator?

Reply:
Ya, integer -1 bitLength is 1. But since the `Random()` generator only generates positive numbers, I think it should be fine right? Just need to add a check that thrown exception when max is negative.

https://api.flutter.dev/flutter/dart-math/Random-class.html